From 2831f4c59cc13ccaeaa7ec6794bf50104f4ed377 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:55 +0800
Subject: [PATCH 2992/4585] [Backport] sw64: kvm: remove MAX_VPN

Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

Reference: https://gitee.com/openeuler/kernel/commit/6be7f996929d3e1b4730e3aa38d09caf0f8b7bd3

--------------------------------

There's no need to manually define MAX_VPN. Use calculated
HARDWARE_VPN_MASK instead.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 arch/sw_64/kvm/kvm-sw64.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/sw_64/kvm/kvm-sw64.c b/arch/sw_64/kvm/kvm-sw64.c
index 1481c3dbb211..e1f7a82fbc26 100644
--- a/arch/sw_64/kvm/kvm-sw64.c
+++ b/arch/sw_64/kvm/kvm-sw64.c
@@ -37,7 +37,6 @@ unsigned long sw64_kvm_last_vpn[NR_CPUS];
 #define cpu_last_vpn(cpuid) sw64_kvm_last_vpn[cpuid]
 
 #ifdef CONFIG_SUBARCH_C3B
-#define MAX_VPN			255
 #define WIDTH_HARDWARE_VPN	8
 #endif
 
@@ -80,7 +79,7 @@ static unsigned long __get_new_vpn_context(struct kvm_vcpu *vcpu, long cpu)
 	unsigned long vpn = cpu_last_vpn(cpu);
 	unsigned long next = vpn + 1;
 
-	if ((vpn & HARDWARE_VPN_MASK) >= MAX_VPN) {
+	if ((vpn & HARDWARE_VPN_MASK) >= HARDWARE_VPN_MASK) {
 		tbia();
 		next = (vpn & ~HARDWARE_VPN_MASK) + VPN_FIRST_VERSION + 1; /* bypass 0 */
 	}
-- 
2.17.1

